pkg/deploy/deployment.go

```diff
@@ -111,10 +126,102 @@ func GetClusterDeployment(name string, namespace string, client runtimeClient.Cl
 	}
 	err := client.Get(context.TODO(), namespacedName, deployment)
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if k8serrors.IsNotFound(err) {
 			return nil, nil
 		}
 		return nil, err
 	}
 	return deployment, nil
 }
+
+func MountSecrets(specDeployment *appsv1.Deployment, deployContext *DeployContext) error {
+	secrets := &corev1.SecretList{}
+
+	kubernetesPartOfLabelSelectorRequirement, _ := labels.NewRequirement(KubernetesPartOfLabelKey, selection.Equals, []string{CheEclipseOrg})
+	kubernetesComponentLabelSelectorRequirement, _ := labels.NewRequirement(KubernetesComponentLabelKey, selection.Equals, []string{specDeployment.Name + "-secret"})
+
+	listOptions := &client.ListOptions{
+		LabelSelector: labels.NewSelector().Add(*kubernetesPartOfLabelSelectorRequirement).Add(*kubernetesComponentLabelSelectorRequirement),
+	}
+	if err := deployContext.ClusterAPI.Client.List(context.TODO(), secrets, listOptions); err != nil {
+		return err
+	}
+
+	// sort secret first to have the same order every time
+	sort.Slice(secrets.Items, func(i, j int) bool {
+		return strings.Compare(secrets.Items[i].Name, secrets.Items[j].Name) < 0
+	})
+
+	for _, secretObj := range secrets.Items {
+		switch secretObj.Annotations[CheEclipseOrgMountAs] {
+		case "file":
+			voluseSource := corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secretObj.Name,
+				},
+			}
+
+			volume := corev1.Volume{
+				Name:         secretObj.Name,
+				VolumeSource: voluseSource,
+			}
+
+			volumeMount := corev1.VolumeMount{
+				Name:      secretObj.Name,
+				MountPath: secretObj.Annotations[CheEclipseOrgMountPath],
+			}
+
+			specDeployment.Spec.Template.Spec.Volumes = append(specDeployment.Spec.Template.Spec.Volumes, volume)
+			specDeployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(specDeployment.Spec.Template.Spec.Containers[0].VolumeMounts, volumeMount)
+
+		case "env":
+			secret, err := GetClusterSecret(secretObj.Name, deployContext.CheCluster.Namespace, deployContext.ClusterAPI)
+			if err != nil {
+				return err
+			} else if secret == nil {
+				return errors.New("Secret " + secretObj.Name + " not found.")
```

Comment:
Please consider using of `fmt.Errorf` instead of `errors.New` here and hereinafter